### PR TITLE
Upgrading to Netty 4.1 and reintroducing proxy support

### PR DESCRIPTION
**Pull request by randak**
Proxy support was removed from an earlier version of the scala-kit due to a lack of stability in Netty 4.1. Now that 4.1 is out, I've reintroduced proxy support. Additionally, I have added it to `previewSession` — it was previously impossible to use preview functionality behind a proxy. A couple of other changes have been made to remove references to deprecated Netty methods.

### CODE REVIEW

File: src/main/scala/io/prismic/HttpClient.scala [hunk 1]
```diff
@@ -71,18 +71,16 @@ object HttpClient {
...
             p.addLast(new HttpProxyHandler(new InetSocketAddress(phost, pport), username, password))
           case Some(ProxyServer(phost, pport, _, _, _, _, _, _)) =>
             p.addLast(new HttpProxyHandler(new InetSocketAddress(phost, pport)))
           case _ => ()
         }
-          */
 
         if (scheme == "https") {
-          val sslCtx = SslContext.newClientContext(SslContext.defaultClientProvider())
+          val sslCtx = SslContextBuilder.forClient().sslProvider(SslContext.defaultClientProvider()).build()
```
Comment (randak) [inline src/main/scala/io/prismic/HttpClient.scala]: `SslContext.newClientContext` is deprecated.

File: src/main/scala/io/prismic/HttpClient.scala [hunk 2]
```diff
@@ -95,13 +93,13 @@ object HttpClient {
             msg match {
               case response: FullHttpResponse =>
                 result.success(ClientResponse(
-                  response.getStatus,
+                  response.status(),
```
Comment (randak) [inline src/main/scala/io/prismic/HttpClient.scala]: `response.getStatus` is deprecated.

File: src/main/scala/io/prismic/Api.scala [hunk 3]
```diff
@@ -43,10 +43,10 @@ final class Api(
    *                (usually the home page of your site)
    * @return a Future corresponding to the URL you should redirect the user to preview the requested change
    */
-  def previewSession(token: String, linkResolver: DocumentLinkResolver, defaultUrl: String): Future[String] = {
+  def previewSession(token: String, linkResolver: DocumentLinkResolver, defaultUrl: String, proxy: Option[ProxyServer] = None): Future[String] = {
     try {
       (for {
-        tokenJson <- HttpClient.getJson(token).map(_.json)
+        tokenJson <- HttpClient.getJson(token, proxy = proxy).map(_.json)
```
Comment (randak) [inline src/main/scala/io/prismic/Api.scala]: This matches the proxy code for `Api.get`

**Comment by randak** [comment]
Looks like the tests are failing due to the removal of the LBC repository they're dependent on. What is a suitable replacement for these?

**Comment by erwan** [comment]
Thank you!

I'll adapt the tests with a new repository so we can make sure they pass for your PR.

**Comment by erwan** [comment]
Hi @randak ,

The tests are fixed on master, could you rebase you branch?

**Comment by randak** [comment]
Going to make a new PR with correct history

**Pull request closed without merging**